modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -233,13 +247,91 @@ private void tryRemovePreviousWritesIndex(RowId rowId, BinaryRow previousRow) {
         }
     }
 
+    /**
+     * Handles the read of a write-intent.
+     *
+     * @param txId Transaction id.
+     * @param rowId Row id.
+     */
+    public void handleWriteIntentRead(UUID txId, RowId rowId) {
```

Comment:
If it's used only within full partition scan on start, please mark given method as deprecated and add following TODO: IGNITE-18502